src/resource.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+# Copyright 2021 Martin Kalcok
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Abstraction of Kubernetes Resources.
+
+This module contains classes that wrap kubernetes resources and enable unified
+interface for their removal.
+"""
+# Remove this pylint skip when functionality is added to `Resource` class
+# pylint: disable=too-few-public-methods
+from typing import Callable
+
+from kubernetes.client import AppsV1Api, CoreV1Api
+from kubernetes.client import RbacAuthorizationV1Api as RbacAuthApi
+from kubernetes.client import StorageV1Api
+
+
+class MissingMethod(BaseException):
+    """Exception that represents method that is not implemented."""
+
+
+class Resource:
```

Comment:
Should this be called `KubernetesResourceBase`?

Reply:
It's a mouthful, but I don't mind. Do you also want to change naming for specific resources, like `Secret` -> `KubernetesSecret` ? Because I think that might be too much. If you'll have something like
```python
from kubernetes_resources import Secret
```
I think it gives enough context.

Reply:
you're raising an interesting point. I do agree that `Secret` could be left as is. A charm developer could use `import Secret as KubernetesSecret` if another `Secret` definition could exist in the future.